olc_art.cpp

```diff
@@ -1296,7 +1246,489 @@ void olc_db::dump(std::ostream &os) const {
 #else
   os << "olc_db dump\n";
 #endif  // UNODB_DETAIL_WITH_STATS
-  olc_art_policy::dump_node(os, root.load());
+  detail::olc_art_policy::dump_node(os, root.load());
+}
+
+///
+/// unodb::olc_db::iterator
+///
+
+// FIXME It is difficult to write this cleanly since NP, KB, CI,
+// etc. are in art_internal_impl.hpp and we are trying to make a tuple
+// declared in a private type for the olc_db::iterator.
+#define make_stack_entry( t, node_critical_section) \
```

Comment:
Please no macros, use a function

Reply:
Resolved the mutex_db::scan() comment.

For make_stack_entry(), I plan to push the push() and pop() into the iterator since we will need to append to the key buffer on push and truncate it on pop.

Reply:
Pushed the changes to take out the macro and encapsulate key buffer management on the iterator (when we have variable length keys).